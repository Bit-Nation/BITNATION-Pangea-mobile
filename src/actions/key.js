// @flow

import { type Mnemonic } from '../types/Mnemonic';

type ValidateEnteredMnemonicAction = { +type: 'VALIDATE_ENTERED_MNEMONIC' };
type ChangeMnemonicValidAction = { +type: 'CHANGE_MNEMONIC_VALID', +mnemonicValid: boolean | null };
type ChangeEnteredMnemonicAction = { +type: 'CHANGE_ENTERED_MNEMONIC', +mnemonic: Mnemonic };

export type Action =
  | ValidateEnteredMnemonicAction
  | ChangeMnemonicValidAction
  | ChangeEnteredMnemonicAction;

export const VALIDATE_ENTERED_MNEMONIC = 'VALIDATE_ENTERED_MNEMONIC';
export const CHANGE_MNEMONIC_VALID = 'CHANGE_MNEMONIC_VALID';
export const CHANGE_ENTERED_MNEMONIC = 'CHANGE_ENTERED_MNEMONIC';

/**
 * @desc Action creator for an action that starts entered mnemonic validation process.
 * @returns {ValidateEnteredMnemonicAction} An action.
 */
export function validateEnteredMnemonic(): ValidateEnteredMnemonicAction {
  return {
    type: VALIDATE_ENTERED_MNEMONIC,
  };
}

/**
 * @desc Action creator for an action that should be called once
 * mnemonic validation status is changed.
 * @param {(boolean|null)} valid Flag that shows if mnemonic is valid.
 * Null is used to mark an undefined state of validation (while it is in progress).
 * @returns {ChangeMnemonicValidAction} An action.
 */
export function changeMnemonicValid(valid: boolean | null): ChangeMnemonicValidAction {
  return {
    type: CHANGE_MNEMONIC_VALID,
    mnemonicValid: valid,
  };
}

/**
 * @desc Action creator for an action that should be called once entered mnemonic is changed.
 * @param {Mnemonic} mnemonic Mnemonic that is entered.
 * @returns {ChangeEnteredMnemonicAction} An action.
 */
export function changeEnteredMnemonic(mnemonic: Mnemonic): ChangeEnteredMnemonicAction {
  return {
    type: CHANGE_ENTERED_MNEMONIC,
    mnemonic,
  };
}
