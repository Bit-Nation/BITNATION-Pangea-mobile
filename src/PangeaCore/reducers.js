// @flow

import { combineReducers } from 'redux';
import accounts, { type State as AccountsState } from '@pangea/accounts/accounts-reducers';
import nations, { type State as NationsState } from '@pangea/nations/nations-reducers';
import modifyNation, { type State as ModifyNationState } from '@pangea/nations/modifyNation-reducers';
import wallet, { type State as WalletState } from '@pangea/wallet/wallet-reducers';
import key, { type State as KeyState } from '@pangea/key/key-reducers';
import activity, { type State as ActivityState } from '@pangea/activity/activity-reducers';
import testingMode, { type State as TestingModeState } from '@pangea/accounts/testingMode-reducers';
import chat, { type State as ChatState } from '@pangea/chat/chat-reducers';
import settings, { type State as SettingsState } from '@pangea/settings/settings-reducers';
import dApps, { type State as DAppsState } from '@pangea/dApps/dApps-reducers';
import documents, { type State as DocumentsState } from '@pangea/documents/documents-reducers';
import contacts, { type State as ContactsState } from '@pangea/contacts/contacts-reducers';

export type State = {
  accounts: AccountsState,
  nations: NationsState,
  modifyNation: ModifyNationState,
  wallet: WalletState,
  key: KeyState,
  activity: ActivityState,
  testingMode: TestingModeState,
  chat: ChatState,
  settings: SettingsState,
  dApps: DAppsState,
  documents: DocumentsState,
  contacts: ContactsState,
}

export const subReducers = {
  accounts,
  nations,
  modifyNation,
  wallet,
  key,
  activity,
  testingMode,
  chat,
  settings,
  dApps,
  documents,
  contacts,
};

export default combineReducers(subReducers);
