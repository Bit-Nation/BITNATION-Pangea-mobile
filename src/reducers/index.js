// @flow

import { combineReducers } from 'redux';
import accounts, { type State as AccountsState } from './accounts';
import nations, { type State as NationsState } from './nations';
import modifyNation, { type State as ModifyNationState } from './modifyNation';
import wallet, { type State as WalletState } from './wallet';
import key, { type State as KeyState } from './key';
import activity, { type State as ActivityState } from './activity';
import testingMode, { type State as TestingModeState } from './testingMode';
import chat, { type State as ChatState } from './chat';
import settings, { type State as SettingsState } from './settings';

export type State = {
  accounts: AccountsState,
  nations: NationsState,
  modifyNation: ModifyNationState,
  wallet: WalletState,
  key: KeyState,
  activity: ActivityState,
  testingMode: TestingModeState,
  chat: ChatState,
  settings: SettingsState,
}

export const subReducers = {
  accounts,
  nations,
  modifyNation,
  wallet,
  key,
  activity,
  testingMode,
  chat,
  settings,
};

export default combineReducers(subReducers);
