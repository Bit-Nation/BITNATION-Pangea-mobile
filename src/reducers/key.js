// @flow

import {
  type Action,
  CHANGE_ENTERED_MNEMONIC,
  CHANGE_MNEMONIC_VALID,
  VALIDATE_ENTERED_MNEMONIC,
} from '../actions/key';
import type { Mnemonic } from '../types/Mnemonic';
import { SERVICES_DESTROYED } from '../actions/serviceContainer';

export type State = {
  +enteredMnemonic: Mnemonic | null,
  +mnemonicValid: boolean | null,
  +mnemonicValidationInProgress: boolean,
}

export const initialState: State = {
  enteredMnemonic: null,
  mnemonicValid: null,
  mnemonicValidationInProgress: false,
};

/**
 * @desc Key reducer.
 * @param {State} state Current state.
 * @param {Action} action Performed action.
 * @returns {State} Next state.
 */
export default (state: State = initialState, action: Action): State => {
  switch (action.type) {
    case SERVICES_DESTROYED:
      return initialState;
    case CHANGE_ENTERED_MNEMONIC:
      return { ...state, enteredMnemonic: action.mnemonic };
    case VALIDATE_ENTERED_MNEMONIC:
      return { ...state, mnemonicValid: null, mnemonicValidationInProgress: true };
    case CHANGE_MNEMONIC_VALID:
      return { ...state, mnemonicValid: action.mnemonicValid, mnemonicValidationInProgress: false };
    default:
      return state;
  }
};

