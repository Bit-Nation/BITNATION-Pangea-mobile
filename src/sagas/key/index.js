import { all, takeEvery } from 'redux-saga/effects';

import { CREATE_PRIVATE_KEY, SAVE_PRIVATE_KEY } from '../../actions/key';
import { createPrivateKeySaga, savePrivateKeySaga } from './sagas';


function* watchCreateWallet() {
  yield takeEvery(CREATE_PRIVATE_KEY, createPrivateKeySaga);
}

function* watchSavePrivateKey() {
  yield takeEvery(SAVE_PRIVATE_KEY, savePrivateKeySaga);
}

export default function* rootSaga() {
  yield all([
    watchCreateWallet(),
    watchSavePrivateKey(),
  ]);
}