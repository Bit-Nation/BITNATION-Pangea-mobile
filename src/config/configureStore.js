// @flow

import { createStore, applyMiddleware, compose, type Store } from 'redux';
import logger from 'redux-logger';
import createSagaMiddleware from 'redux-saga';
import createSagaMonitor from '@clarketm/saga-monitor';
import reducers from '../reducers';
import rootSaga from '../sagas';

/**
 * @desc Configures a Redux store.
 * @return {Store} Created store object.
 */
export default function configureStore(): Store {
  const sagaMonitor = createSagaMonitor({
    level: 'log',
    actionDispatch: true,
  });
  const sagaMiddleware = createSagaMiddleware({ sagaMonitor });

  const enhancer = compose(
    applyMiddleware(sagaMiddleware),
    applyMiddleware(logger),
  );

  const store = createStore(reducers, enhancer);
  sagaMiddleware.run(rootSaga);
  return store;
}
