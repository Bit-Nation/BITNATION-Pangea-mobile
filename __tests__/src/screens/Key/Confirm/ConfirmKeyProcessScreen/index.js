import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';

import { initialState } from '../../../../../../src/reducers/key';
import ConfirmKeyProcessScreen from '../../../../../../src/screens/Key/Confirm/ConfirmKeyProcessScreen';
import navigatorMock from '../../../../../../__mocks__/Navigator';

test('ConfirmKeyProcessScreen renders correctly', () => {
  const initialStateMock = {
    key: initialState,
  };
  const storeMock = configureStore([]);
  const propsMock = {
    navigator: navigatorMock,
  };

  const wrapper = shallow((
    <ConfirmKeyProcessScreen {...propsMock} store={storeMock(initialStateMock)} />
  ));
  expect(wrapper).toMatchSnapshot();
});
