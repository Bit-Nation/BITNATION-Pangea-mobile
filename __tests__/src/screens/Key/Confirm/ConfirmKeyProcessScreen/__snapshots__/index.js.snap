// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmKeyProcessScreen renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ConfirmKeyProcessScreen
    navigator={
      Object {
        "pop": [MockFunction],
        "push": [MockFunction],
        "setButtons": [MockFunction],
        "setOnNavigatorEvent": [MockFunction] {
          "calls": Array [
            Array [
              [Function],
            ],
          ],
        },
      }
    }
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <BackgroundImage
          source={1}
          style={undefined}
        />,
        <FakeNavigationBar
          navBarHidden={false}
        />,
        <View
          style={
            Object {
              "alignContent": "flex-start",
              "flex": 1,
              "flexDirection": "column",
              "justifyContent": "flex-start",
              "marginLeft": 8,
              "marginRight": 8,
            }
          }
        >
          <View
            style={
              Object {
                "height": 18,
              }
            }
          />
          <PanelView
            body={undefined}
            buttonTitle="Button"
            childrenContainerStyle={
              Object {
                "flex": 0,
              }
            }
            icon={undefined}
            onButtonClick={null}
            renderAdditionalInfo={null}
            renderBottom={null}
            style={
              Object {
                "backgroundColor": "transparent",
                "borderRadius": 8,
                "flexDirection": "column",
                "justifyContent": "space-between",
                "marginBottom": 8,
                "marginLeft": 0,
                "marginRight": 0,
                "marginTop": 8,
                "paddingBottom": 4,
                "paddingLeft": 8,
                "paddingRight": 8,
                "paddingTop": 4,
              }
            }
            title={undefined}
            titleStyle={
              Object {
                "color": "white",
                "fontFamily": "Roboto",
                "fontSize": 22,
                "fontWeight": "bold",
                "letterSpacing": 0,
                "lineHeight": 28,
                "textAlign": "left",
              }
            }
          >
            <BodyParagraphs
              paragraphs={
                Array [
                  "Write each row of words on the piece of paper. Press “Next” when you have written the row.",
                ]
              }
            />
            <View
              style={
                Object {
                  "backgroundColor": "rgba(27,57,92,0.7)",
                  "borderRadius": 8,
                  "height": 241,
                  "paddingBottom": 3,
                  "paddingLeft": 12,
                  "paddingRight": 12,
                  "paddingTop": 20,
                }
              }
            >
              <GridView
                activeRow={0}
                disableInactiveRows={true}
                itemsPerRow={3}
                renderItem={[Function]}
                rowsCount={4}
                style={
                  Object {
                    "flex": 1,
                  }
                }
              />
            </View>
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "marginLeft": 13,
                  "marginRight": 13,
                  "marginTop": 15,
                }
              }
            >
              <Button
                enabled={false}
                onPress={[Function]}
                style={
                  Object {
                    "width": 90,
                  }
                }
                title="Previous"
              />
              <Button
                enabled={true}
                onPress={[Function]}
                style={
                  Object {
                    "width": 90,
                  }
                }
                title="Next"
              />
            </View>
          </PanelView>
        </View>,
      ],
      "style": Object {
        "alignItems": "stretch",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "space-between",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "source": 1,
          "style": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "navBarHidden": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <View
              style={
                Object {
                  "height": 18,
                }
              }
            />,
            <PanelView
              body={undefined}
              buttonTitle="Button"
              childrenContainerStyle={
                Object {
                  "flex": 0,
                }
              }
              icon={undefined}
              onButtonClick={null}
              renderAdditionalInfo={null}
              renderBottom={null}
              style={
                Object {
                  "backgroundColor": "transparent",
                  "borderRadius": 8,
                  "flexDirection": "column",
                  "justifyContent": "space-between",
                  "marginBottom": 8,
                  "marginLeft": 0,
                  "marginRight": 0,
                  "marginTop": 8,
                  "paddingBottom": 4,
                  "paddingLeft": 8,
                  "paddingRight": 8,
                  "paddingTop": 4,
                }
              }
              title={undefined}
              titleStyle={
                Object {
                  "color": "white",
                  "fontFamily": "Roboto",
                  "fontSize": 22,
                  "fontWeight": "bold",
                  "letterSpacing": 0,
                  "lineHeight": 28,
                  "textAlign": "left",
                }
              }
            >
              <BodyParagraphs
                paragraphs={
                  Array [
                    "Write each row of words on the piece of paper. Press “Next” when you have written the row.",
                  ]
                }
              />
              <View
                style={
                  Object {
                    "backgroundColor": "rgba(27,57,92,0.7)",
                    "borderRadius": 8,
                    "height": 241,
                    "paddingBottom": 3,
                    "paddingLeft": 12,
                    "paddingRight": 12,
                    "paddingTop": 20,
                  }
                }
              >
                <GridView
                  activeRow={0}
                  disableInactiveRows={true}
                  itemsPerRow={3}
                  renderItem={[Function]}
                  rowsCount={4}
                  style={
                    Object {
                      "flex": 1,
                    }
                  }
                />
              </View>
              <View
                style={
                  Object {
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                    "marginLeft": 13,
                    "marginRight": 13,
                    "marginTop": 15,
                  }
                }
              >
                <Button
                  enabled={false}
                  onPress={[Function]}
                  style={
                    Object {
                      "width": 90,
                    }
                  }
                  title="Previous"
                />
                <Button
                  enabled={true}
                  onPress={[Function]}
                  style={
                    Object {
                      "width": 90,
                    }
                  }
                  title="Next"
                />
              </View>
            </PanelView>,
          ],
          "style": Object {
            "alignContent": "flex-start",
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "flex-start",
            "marginLeft": 8,
            "marginRight": 8,
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "style": Object {
                "height": 18,
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "body": undefined,
              "buttonTitle": "Button",
              "children": Array [
                <BodyParagraphs
                  paragraphs={
                    Array [
                      "Write each row of words on the piece of paper. Press “Next” when you have written the row.",
                    ]
                  }
                />,
                <View
                  style={
                    Object {
                      "backgroundColor": "rgba(27,57,92,0.7)",
                      "borderRadius": 8,
                      "height": 241,
                      "paddingBottom": 3,
                      "paddingLeft": 12,
                      "paddingRight": 12,
                      "paddingTop": 20,
                    }
                  }
                >
                  <GridView
                    activeRow={0}
                    disableInactiveRows={true}
                    itemsPerRow={3}
                    renderItem={[Function]}
                    rowsCount={4}
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  />
                </View>,
                <View
                  style={
                    Object {
                      "flexDirection": "row",
                      "justifyContent": "space-between",
                      "marginLeft": 13,
                      "marginRight": 13,
                      "marginTop": 15,
                    }
                  }
                >
                  <Button
                    enabled={false}
                    onPress={[Function]}
                    style={
                      Object {
                        "width": 90,
                      }
                    }
                    title="Previous"
                  />
                  <Button
                    enabled={true}
                    onPress={[Function]}
                    style={
                      Object {
                        "width": 90,
                      }
                    }
                    title="Next"
                  />
                </View>,
              ],
              "childrenContainerStyle": Object {
                "flex": 0,
              },
              "icon": undefined,
              "onButtonClick": null,
              "renderAdditionalInfo": null,
              "renderBottom": null,
              "style": Object {
                "backgroundColor": "transparent",
                "borderRadius": 8,
                "flexDirection": "column",
                "justifyContent": "space-between",
                "marginBottom": 8,
                "marginLeft": 0,
                "marginRight": 0,
                "marginTop": 8,
                "paddingBottom": 4,
                "paddingLeft": 8,
                "paddingRight": 8,
                "paddingTop": 4,
              },
              "title": undefined,
              "titleStyle": Object {
                "color": "white",
                "fontFamily": "Roboto",
                "fontSize": 22,
                "fontWeight": "bold",
                "letterSpacing": 0,
                "lineHeight": 28,
                "textAlign": "left",
              },
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "paragraphs": Array [
                    "Write each row of words on the piece of paper. Press “Next” when you have written the row.",
                  ],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <GridView
                    activeRow={0}
                    disableInactiveRows={true}
                    itemsPerRow={3}
                    renderItem={[Function]}
                    rowsCount={4}
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  />,
                  "style": Object {
                    "backgroundColor": "rgba(27,57,92,0.7)",
                    "borderRadius": 8,
                    "height": 241,
                    "paddingBottom": 3,
                    "paddingLeft": 12,
                    "paddingRight": 12,
                    "paddingTop": 20,
                  },
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "activeRow": 0,
                    "disableInactiveRows": true,
                    "itemsPerRow": 3,
                    "renderItem": [Function],
                    "rowsCount": 4,
                    "style": Object {
                      "flex": 1,
                    },
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    <Button
                      enabled={false}
                      onPress={[Function]}
                      style={
                        Object {
                          "width": 90,
                        }
                      }
                      title="Previous"
                    />,
                    <Button
                      enabled={true}
                      onPress={[Function]}
                      style={
                        Object {
                          "width": 90,
                        }
                      }
                      title="Next"
                    />,
                  ],
                  "style": Object {
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                    "marginLeft": 13,
                    "marginRight": 13,
                    "marginTop": 15,
                  },
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "children": undefined,
                      "enabled": false,
                      "onPress": [Function],
                      "style": Object {
                        "width": 90,
                      },
                      "title": "Previous",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "children": undefined,
                      "enabled": true,
                      "onPress": [Function],
                      "style": Object {
                        "width": 90,
                      },
                      "title": "Next",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <BackgroundImage
            source={1}
            style={undefined}
          />,
          <FakeNavigationBar
            navBarHidden={false}
          />,
          <View
            style={
              Object {
                "alignContent": "flex-start",
                "flex": 1,
                "flexDirection": "column",
                "justifyContent": "flex-start",
                "marginLeft": 8,
                "marginRight": 8,
              }
            }
          >
            <View
              style={
                Object {
                  "height": 18,
                }
              }
            />
            <PanelView
              body={undefined}
              buttonTitle="Button"
              childrenContainerStyle={
                Object {
                  "flex": 0,
                }
              }
              icon={undefined}
              onButtonClick={null}
              renderAdditionalInfo={null}
              renderBottom={null}
              style={
                Object {
                  "backgroundColor": "transparent",
                  "borderRadius": 8,
                  "flexDirection": "column",
                  "justifyContent": "space-between",
                  "marginBottom": 8,
                  "marginLeft": 0,
                  "marginRight": 0,
                  "marginTop": 8,
                  "paddingBottom": 4,
                  "paddingLeft": 8,
                  "paddingRight": 8,
                  "paddingTop": 4,
                }
              }
              title={undefined}
              titleStyle={
                Object {
                  "color": "white",
                  "fontFamily": "Roboto",
                  "fontSize": 22,
                  "fontWeight": "bold",
                  "letterSpacing": 0,
                  "lineHeight": 28,
                  "textAlign": "left",
                }
              }
            >
              <BodyParagraphs
                paragraphs={
                  Array [
                    "Write each row of words on the piece of paper. Press “Next” when you have written the row.",
                  ]
                }
              />
              <View
                style={
                  Object {
                    "backgroundColor": "rgba(27,57,92,0.7)",
                    "borderRadius": 8,
                    "height": 241,
                    "paddingBottom": 3,
                    "paddingLeft": 12,
                    "paddingRight": 12,
                    "paddingTop": 20,
                  }
                }
              >
                <GridView
                  activeRow={0}
                  disableInactiveRows={true}
                  itemsPerRow={3}
                  renderItem={[Function]}
                  rowsCount={4}
                  style={
                    Object {
                      "flex": 1,
                    }
                  }
                />
              </View>
              <View
                style={
                  Object {
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                    "marginLeft": 13,
                    "marginRight": 13,
                    "marginTop": 15,
                  }
                }
              >
                <Button
                  enabled={false}
                  onPress={[Function]}
                  style={
                    Object {
                      "width": 90,
                    }
                  }
                  title="Previous"
                />
                <Button
                  enabled={true}
                  onPress={[Function]}
                  style={
                    Object {
                      "width": 90,
                    }
                  }
                  title="Next"
                />
              </View>
            </PanelView>
          </View>,
        ],
        "style": Object {
          "alignItems": "stretch",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "space-between",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "source": 1,
            "style": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "navBarHidden": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <View
                style={
                  Object {
                    "height": 18,
                  }
                }
              />,
              <PanelView
                body={undefined}
                buttonTitle="Button"
                childrenContainerStyle={
                  Object {
                    "flex": 0,
                  }
                }
                icon={undefined}
                onButtonClick={null}
                renderAdditionalInfo={null}
                renderBottom={null}
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "borderRadius": 8,
                    "flexDirection": "column",
                    "justifyContent": "space-between",
                    "marginBottom": 8,
                    "marginLeft": 0,
                    "marginRight": 0,
                    "marginTop": 8,
                    "paddingBottom": 4,
                    "paddingLeft": 8,
                    "paddingRight": 8,
                    "paddingTop": 4,
                  }
                }
                title={undefined}
                titleStyle={
                  Object {
                    "color": "white",
                    "fontFamily": "Roboto",
                    "fontSize": 22,
                    "fontWeight": "bold",
                    "letterSpacing": 0,
                    "lineHeight": 28,
                    "textAlign": "left",
                  }
                }
              >
                <BodyParagraphs
                  paragraphs={
                    Array [
                      "Write each row of words on the piece of paper. Press “Next” when you have written the row.",
                    ]
                  }
                />
                <View
                  style={
                    Object {
                      "backgroundColor": "rgba(27,57,92,0.7)",
                      "borderRadius": 8,
                      "height": 241,
                      "paddingBottom": 3,
                      "paddingLeft": 12,
                      "paddingRight": 12,
                      "paddingTop": 20,
                    }
                  }
                >
                  <GridView
                    activeRow={0}
                    disableInactiveRows={true}
                    itemsPerRow={3}
                    renderItem={[Function]}
                    rowsCount={4}
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  />
                </View>
                <View
                  style={
                    Object {
                      "flexDirection": "row",
                      "justifyContent": "space-between",
                      "marginLeft": 13,
                      "marginRight": 13,
                      "marginTop": 15,
                    }
                  }
                >
                  <Button
                    enabled={false}
                    onPress={[Function]}
                    style={
                      Object {
                        "width": 90,
                      }
                    }
                    title="Previous"
                  />
                  <Button
                    enabled={true}
                    onPress={[Function]}
                    style={
                      Object {
                        "width": 90,
                      }
                    }
                    title="Next"
                  />
                </View>
              </PanelView>,
            ],
            "style": Object {
              "alignContent": "flex-start",
              "flex": 1,
              "flexDirection": "column",
              "justifyContent": "flex-start",
              "marginLeft": 8,
              "marginRight": 8,
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "style": Object {
                  "height": 18,
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "body": undefined,
                "buttonTitle": "Button",
                "children": Array [
                  <BodyParagraphs
                    paragraphs={
                      Array [
                        "Write each row of words on the piece of paper. Press “Next” when you have written the row.",
                      ]
                    }
                  />,
                  <View
                    style={
                      Object {
                        "backgroundColor": "rgba(27,57,92,0.7)",
                        "borderRadius": 8,
                        "height": 241,
                        "paddingBottom": 3,
                        "paddingLeft": 12,
                        "paddingRight": 12,
                        "paddingTop": 20,
                      }
                    }
                  >
                    <GridView
                      activeRow={0}
                      disableInactiveRows={true}
                      itemsPerRow={3}
                      renderItem={[Function]}
                      rowsCount={4}
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    />
                  </View>,
                  <View
                    style={
                      Object {
                        "flexDirection": "row",
                        "justifyContent": "space-between",
                        "marginLeft": 13,
                        "marginRight": 13,
                        "marginTop": 15,
                      }
                    }
                  >
                    <Button
                      enabled={false}
                      onPress={[Function]}
                      style={
                        Object {
                          "width": 90,
                        }
                      }
                      title="Previous"
                    />
                    <Button
                      enabled={true}
                      onPress={[Function]}
                      style={
                        Object {
                          "width": 90,
                        }
                      }
                      title="Next"
                    />
                  </View>,
                ],
                "childrenContainerStyle": Object {
                  "flex": 0,
                },
                "icon": undefined,
                "onButtonClick": null,
                "renderAdditionalInfo": null,
                "renderBottom": null,
                "style": Object {
                  "backgroundColor": "transparent",
                  "borderRadius": 8,
                  "flexDirection": "column",
                  "justifyContent": "space-between",
                  "marginBottom": 8,
                  "marginLeft": 0,
                  "marginRight": 0,
                  "marginTop": 8,
                  "paddingBottom": 4,
                  "paddingLeft": 8,
                  "paddingRight": 8,
                  "paddingTop": 4,
                },
                "title": undefined,
                "titleStyle": Object {
                  "color": "white",
                  "fontFamily": "Roboto",
                  "fontSize": 22,
                  "fontWeight": "bold",
                  "letterSpacing": 0,
                  "lineHeight": 28,
                  "textAlign": "left",
                },
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "paragraphs": Array [
                      "Write each row of words on the piece of paper. Press “Next” when you have written the row.",
                    ],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <GridView
                      activeRow={0}
                      disableInactiveRows={true}
                      itemsPerRow={3}
                      renderItem={[Function]}
                      rowsCount={4}
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    />,
                    "style": Object {
                      "backgroundColor": "rgba(27,57,92,0.7)",
                      "borderRadius": 8,
                      "height": 241,
                      "paddingBottom": 3,
                      "paddingLeft": 12,
                      "paddingRight": 12,
                      "paddingTop": 20,
                    },
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "activeRow": 0,
                      "disableInactiveRows": true,
                      "itemsPerRow": 3,
                      "renderItem": [Function],
                      "rowsCount": 4,
                      "style": Object {
                        "flex": 1,
                      },
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": Array [
                      <Button
                        enabled={false}
                        onPress={[Function]}
                        style={
                          Object {
                            "width": 90,
                          }
                        }
                        title="Previous"
                      />,
                      <Button
                        enabled={true}
                        onPress={[Function]}
                        style={
                          Object {
                            "width": 90,
                          }
                        }
                        title="Next"
                      />,
                    ],
                    "style": Object {
                      "flexDirection": "row",
                      "justifyContent": "space-between",
                      "marginLeft": 13,
                      "marginRight": 13,
                      "marginTop": 15,
                    },
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "children": undefined,
                        "enabled": false,
                        "onPress": [Function],
                        "style": Object {
                          "width": 90,
                        },
                        "title": "Previous",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "children": undefined,
                        "enabled": true,
                        "onPress": [Function],
                        "style": Object {
                          "width": 90,
                        },
                        "title": "Next",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                  ],
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
